lib/experimental/Collections/Card/index.tsx

```diff
@@ -20,45 +22,75 @@ export const CardCollection = <
   title,
   source,
 }: CollectionProps<Record, Filters, CardVisualizationOptions<Record>>) => {
-  const { data, isLoading } = useData<Record, Filters>(source)
+  // We override this to force a perPage of 24 (unless set at the dataAdapter
+  // level), which is a multiple of 2, 3 and 4 This is to ensure that the cards
+  // are always aligned in a grid
+  const overridenSource = useMemo(() => {
+    if (source.dataAdapter.paginationType === "pages") {
+      return {
+        ...source,
+        dataAdapter: {
+          ...source.dataAdapter,
+          perPage: source.dataAdapter.perPage ?? 24,
+        },
+      }
+    }
+    return source
+  }, [source])
+
+  const { data, paginationInfo, setPage, isInitialLoading } = useData<
+    Record,
+    Filters
+  >(overridenSource)
 
   return (
-    <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4">
-      {isLoading
-        ? Array.from({ length: 8 }).map((_, i) => (
-            <Card key={i}>
-              <CardHeader>
-                <CardTitle>
-                  <Skeleton className="h-4 w-3/4" />
-                </CardTitle>
-              </CardHeader>
-              <CardContent className="space-y-2">
-                {cardProperties.map((property) => (
-                  <div key={String(property.label)} className="space-y-1">
-                    <Skeleton className="h-3 w-1/4" />
-                    <Skeleton className="h-3 w-1/2" />
-                  </div>
-                ))}
-              </CardContent>
-            </Card>
-          ))
-        : data.map((item, index) => (
-            <Card key={index}>
-              <CardHeader>
-                <CardTitle>{title(item)}</CardTitle>
-              </CardHeader>
-              <CardContent className="space-y-2">
-                {cardProperties.map((property) => (
-                  <div key={String(property.label)} className="space-y-1">
-                    <div className="text-muted-foreground text-sm font-medium">
-                      {property.label}
+    <>
+      <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4">
+        {isInitialLoading
```

Comment:
I don't understand the difference between `isInitialLoading` and `isLoading`.

Reply:
This might be a bit overengineered, but in my mind it makes sense. I found it awkward that the loading between pages would show the skeleton, as it would move the layout a bit. Imagine clicking on a pagination page and getting your table smaller and then bigger. 

So I thought it might be a good idea to have two loading states and let each implementation decide how to show feedback. In this case, we would show a skeleton on initial load, and then just show a spinner somewhere else to indicate data is being fetched. Does that sound reasonable?

Reply:
IF you want to avoid the height changes there are 2 possible solutions i uses before (i could be more)
Define a default height for the table and let developers to set it if they need another value (and have the option to full-height) where the table tries to use all the parent height
Another solution is to get the height value before the first load and then force the table to use it (in any case this still having the issue of height change on the first load)